src/extensions/default/Inline3DParametersEditor/Inline3DParameterEditor.js

```diff
@@ -0,0 +1,201 @@
+/*
+ * File based on InlineColorEditor/InlineColorEditor.js
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var InlineWidget            = brackets.getModule("editor/InlineWidget").InlineWidget,
+        ParameterEditor         = require("ParameterEditor").ParameterEditor,
+        Inline3dParametersUtils = require("Parameters3DUtils");
+
+    /** @type {number} Global var used to provide a unique ID for each parameter editor instance's _origin field. */
+    var lastOriginId = 1;
+
+    /**
+     * Inline widget containing a ParameterEditor control
+     * @param {!string} parameters  Initial Parameters
+     * @param {!CodeMirror.TextMarker} marker
+     */
+    function Inline3DParameterEditor(parameters, marker, tag) {
+        this._parameters = parameters;
+        this._marker = marker;
+        this._tag = tag;
+        this._isOwnChange = false;
+        this._isHostChange = false;
+        this._origin = "+InlineParameterditor_" + (lastOriginId++);
+
+        this._handleParametersChange = this._handleParametersChange.bind(this);
+        this._handleHostDocumentChange = this._handleHostDocumentChange.bind(this);
+        InlineWidget.call(this);
+    }
+
+    Inline3DParameterEditor.prototype = Object.create(InlineWidget.prototype);
+    Inline3DParameterEditor.prototype.constructor = Inline3DParameterEditor;
+    Inline3DParameterEditor.prototype.parentClass = InlineWidget.prototype;
+
+    Inline3DParameterEditor.prototype.ParameterEditor = null;
+
+    Inline3DParameterEditor.prototype._parameters = null;
+
+    /**
+     * Range of code we're attached to; _marker.find() may by null if sync is lost.
+     * @type {!CodeMirror.TextMarker}
+     */
+    Inline3DParameterEditor.prototype._marker = null;
+
+    /** @type {boolean} True while we're syncing a parameter picker change into the code editor */
+    Inline3DParameterEditor.prototype._isOwnChange = null;
+
+    /** @type {boolean} True while we're syncing a code editor change into the parameter picker */
+    Inline3DParameterEditor.prototype._isHostChange = null;
+
+    /** @type {number} ID used to identify edits coming from this inline widget for undo batching */
+    Inline3DParameterEditor.prototype._origin = null;
```

Comment:
Out of curiosity, why do you bother to set all these to `null` on the prototype, since you already have values on the object itself?  What purpose does this serve?